src/sasctl/pzmm/write_json_files.py

```diff
@@ -950,3 +982,629 @@ def convertDataRole(self, dataRole):
+        recommended to the user that they adjust the requirements.json file's package versions
+        to match the model development environment.
+
+        Parameters
+        ----------
+        json_path : str, optional
+            The path to a Python project, by default the current working directory.
+
+        Yields
+        ------
+        requirements.json : file
+            JSON file used to create a specific Python environment in a SAS Model Manager published container.
+        """
+
+        pickle_packages = []
+        pickle_files = cls.get_pickle_file(json_path)
+        for pickle_file in pickle_files:
+            pickle_packages.append(cls.get_pickle_dependencies(pickle_file))
+
+        code_dependencies = cls.get_code_dependencies(json_path)
+
+        package_list = list(pickle_packages) + code_dependencies
+        package_list = list(set(list(flatten(package_list))))
+        package_list = cls.remove_standard_library_packages(package_list)
+        package_and_version = cls.get_local_package_version(package_list)
+        # Identify packages with missing versions
+        missing_package_versions = [
+            item[0] for item in package_and_version if not item[1]
+        ]
+
+        with open(Path(json_path) / "requirements.json", "w") as file:
+            if missing_package_versions:
+                json_step = json.dumps(
+                    [
+                        {
+                            "Warning": "The existence and/or versions for the following packages could not be "
+                            "determined:",
+                            "Packages": ", ".join(missing_package_versions),
+                        }
+                    ],
+                    indent=4,
+                )
+                file.write(json_step)
+
+            for package, version in package_and_version:
+                if version:
+                    json_step = json.dumps(
+                        [
+                            {
+                                "step": "install " + package,
+                                "command": "pip install " + package + "==" + version,
+                            }
+                        ],
+                        indent=4,
+                    )
+                file.write(json_step)
+
+    @classmethod
+    def get_local_package_version(cls, package_list):
+        """
+        Get package_name versions from the local environment. If the package_name
+        does not contain an attribute of "__version__", "version", or
+        "VERSION", no package_name version will be found.
+
+        Parameters
+        ----------
+        package_list : list
+            List of Python packages.
+
+        Returns
+        -------
+        list
+            Nested list of Python package_name names and found versions.
+        """
+
+        def package_not_found_output(package_name, package_versions):
+            print(
```

Comment:
Do not print.  This should either be a warning or a log message.  Using print() prevents the user from controlling if/when such messages are suppressed.

Reply:
Converted to warnings.warn() here and at another location within write_json_files.py, as well as removed redundant try except block also attempting to print a warning.